src/hooks/useWithdrawalInfo.ts

```diff
@@ -0,0 +1,110 @@
+import { useTokenConfigs } from './useTokenConfigs';
+
+const BLOCK_TIME = isMainnet ? 1_000 : 1_500;
+
+export const useWithdrawalInfo = ({ isTransfer }: { isTransfer?: boolean }) => {
+  const { getWithdrawalAndTransferGatingStatus, getWithdrawalCapacityByDenom } = useDydxClient();
+  const { usdcDenom, usdcDecimals } = useTokenConfigs();
+  const apiState = useSelector(getApiState, shallowEqual);
+  const { height } = apiState || {};
+  const selectedLocale = useSelector(getSelectedLocale);
+  const dispatch = useDispatch();
+
+  const { data: usdcWithdawalCapacity } = useQuery({
+    queryKey: 'usdcWithdrawalCapacity',
+    queryFn: async () => {
+      try {
+        const response = await getWithdrawalCapacityByDenom({ denom: usdcDenom });
+        return JSON.parse(encodeJson(response, ByteArrayEncoding.BIGINT));
+      } catch (error) {
+        console.error('error');
+      }
+    },
+    refetchInterval: 60_000,
+    staleTime: 60_000,
+  });
+
+  const { data: withdrawalAndTransferGatingStatus } = useQuery({
+    queryKey: 'withdrawalTransferGateStatus',
+    queryFn: getWithdrawalAndTransferGatingStatus,
+    refetchInterval: 60_000,
+    staleTime: 60_000,
+  });
+
+  const capacity = useMemo(() => {
+    const capacityList = usdcWithdawalCapacity?.limiterCapacityList;
+    if (!capacityList || capacityList.length < 2) {
+      return BIG_NUMBERS.ZERO;
+    }
+
+    const [{ capacity: daily }, { capacity: weekly }] = capacityList;
+    const dailyBN = MustBigNumber(daily);
+    const weeklyBN = MustBigNumber(weekly);
+    return BigNumber.minimum(dailyBN, weeklyBN).div(10 ** usdcDecimals);
+  }, [usdcDecimals, usdcWithdawalCapacity]);
+
+  const withdrawalAndTransferGatingStatusValue = useMemo(() => {
+    const { withdrawalsAndTransfersUnblockedAtBlock } = withdrawalAndTransferGatingStatus ?? {};
+    if (
+      height &&
+      withdrawalsAndTransfersUnblockedAtBlock &&
+      height < withdrawalsAndTransfersUnblockedAtBlock
+    ) {
+      return {
+        estimatedUnblockTime: formatRelativeTime(
+          Date.now() + (withdrawalsAndTransfersUnblockedAtBlock - height) * BLOCK_TIME,
+          {
+            locale: selectedLocale,
+            largestUnit: 'day',
+          }
+        ),
+        isGated: true,
+      };
+    }
+    return {
+      estimatedUnblockTime: null,
+      isGated: false,
+    };
+  }, [height, withdrawalAndTransferGatingStatus]);
+
+  useEffect(() => {
+    if (
+      withdrawalAndTransferGatingStatusValue.isGated &&
+      withdrawalAndTransferGatingStatusValue.estimatedUnblockTime
+    ) {
+      dispatch(closeDialog());
+      dispatch(
+        openDialog({
+          type: DialogTypes.WithdrawalGated,
+          dialogProps: {
+            isTransfer,
+            estimatedUnblockTime: withdrawalAndTransferGatingStatusValue.estimatedUnblockTime,
+          },
+        })
+      );
+    }
+  }, [withdrawalAndTransferGatingStatusValue.isGated]);
```

Comment:
nit: would we need `withdrawalAndTransferGatingStatusValue.estimatedUnblockTime` and `isTransfer` as deps as well?

Reply:
Not estimatedUnblockTime, because it gets updated when block height is updated (polling fn has interval of ~8 seconds). We wouldn't want to trigger the dialog that often.